src/Microsoft.AspNetCore.Mvc.Core/Internal/ObjectResultExecutor.cs

```diff
@@ -131,6 +137,7 @@ public virtual Task ExecuteAsync(ActionContext context, ObjectResult result)
                 // No formatter supports this.
                 Logger.NoFormatter(formatterContext);
 
+                formatterContext.ContentType = default(StringSegment);
```

Comment:
Why is this necessary? Won't be using `formatterContext` when returning a `406` status.